### DISCUSSION TITLE
Insert New Supplier fails if no Supplier Type

### DISCUSSION THREAD

**Discussion by dalers** [timschofield/webERP#249] [Problems / Bugs]
I attempted to create a new supplier

Click [Main Menu > Payables > Maintenance > Add a Supplier]

![image](https://github.com/user-attachments/assets/d89eef77-f9a8-40f0-b6e3-6487e9e5c6f2)

Entered data for new supplier

![image](https://github.com/user-attachments/assets/2450a58e-8281-4908-88be-874a22ff128e)

Clicked [Insert New Supplier], and next page did not render correctly.

![image](https://github.com/user-attachments/assets/471d6729-238e-40a9-aab1-9e4e29868e11)

httpd-error.log contained follow error:

```
[Wed Nov 20 15:38:30.033467 2024] [php:error] [pid 59137] [client 162.223.103.50:49198] PHP Fatal error:  Uncaught mysqli_sql_exception: Incorrect integer value: '' for column `weberp_scc`.`suppliers`.`supptype` at row 1 in /usr/local/www/weberp/includes/ConnectDB_mysqli.inc:58\nStack trace:\n#0 /usr/local/www/weberp/includes/ConnectDB_mysqli.inc(58): mysqli_query()\n#1 /usr/local/www/weberp/Suppliers.php(588): DB_query()\n#2 {main}\n  thrown in /usr/local/www/weberp/includes/ConnectDB_mysqli.inc on line 58, referer: https://weberp.dalescott.net/Suppliers.php
dale@whizzer:~ $
```

```
webERP Version for Issue Report
-------------------------------
Git Branch:
* master

Git Commit:
commit 98ed6e29256f83a099f9f638f36265c4fa00a18d (HEAD -> master, origin/master, origin/HEAD)
Author: Tim Schofield <tim.schofield1960@googlemail.com>
Date:   Fri Nov 15 20:41:19 2024 +0000

    Ensure page is appearing properly

     Files changed in commit:
     doc/Manual/style/manual.css

PHP Version:
PHP 8.1.30 (cli) (built: Oct  8 2024 02:10:22) (NTS)
```

**Comment by dalers** [comment]
I realized from the error message that a Supplier Type must be defined before a Supplier can be created. I created a Supplier Type ("DEFAULT") and was then able to successfully create a supplier. This should be caught before it hits the database. 

Fyi, in the process I found another uncaught error - entering too many characters into Address 2 for a supplier. I don't know the exact maximum length allowed, but seemed to be shorter than the "should be less than 40 characters" given by the tool tip.

**Comment by timschofield** [comment]
Hi Dale, I have put in a check for the supplier types.

The only issues on address line length I found was on line 5 which should
have been 20 not 40 and I have corrected that. Address line 2 has length 40
in the database, and is restricted to 40 characters in the input field. Can
you check that is still a problem?

Thanks
Tim

On Thu, 21 Nov 2024 at 00:09, Dale Scott ***@***.***> wrote:

> I realized from the error message that a Supplier Type must be defined
> before a Supplier can be created. I created a Supplier Type ("DEFAULT") and
> was then able to successfully create a supplier.
>
> This error should be caught before the database.
>
> Fwiw, entering too many characters into Address 2 is also an uncaught
> error. I don't know the exact maximum length, but I found it was shorter
> than the tool tip "should be less than 40 characters".
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/249#discussioncomment-11329773>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LAT2NUVCLNLOFWWE3D2BUQD3AVCNFSM6AAAAABSFU35Q6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZSHE3TOMY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by dalers** [comment]
Problem is resolved. I created a new supplier using maximum number of characters per tip in field without an error.

However, I think the "field tips" (btw, what is the correct name?) would be better if they simply specified the max number of characters insetad of "within" and "less than". What does everyone think? Agree that tips like this can be changed when found?

![image](https://github.com/user-attachments/assets/e02389e5-4e5d-4065-9653-a0d7121bf393)

**Comment by timschofield** [comment]
They are called placeholders. I think these were originally written by
Exson Qu and English isn't his first language, so if they can be
improved just change them and do a pull request. ;)

Thanks
Tim

On Sat, 7 Dec 2024 at 19:02, Dale Scott ***@***.***> wrote:
>
> Problem resolved. I created a new supplier using maximum number of characters per tip in field with noo errors.
>
> However, I think the "field tips" (btw, what is the correct name?) would be better if they simply specified the max number of characters insetad of "within" and "less than". What does everyone think? Agree that tips like this can be changed when found?
>
> image.png (view on web)
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by dalers** [comment]
Issue closed, placeholders updated for consistency by commit df321578384546edf8daf6901a01d4882b031caf
